scientific-computing-with-python/learn-tree-traversal-by-building-a-binary-search-tree/binary-search-tree.py

```diff
@@ -0,0 +1,83 @@
+class TreeNode:
+    def __init__(self, key):
+        self.key = key
+        self.left = None
+        self.right = None
+
+class BinarySearchTree:
+    def __init__(self):
+        self.root = None
+
+    def insert(self, key):
+        self.root = self._insert(self.root, key)
+
+    def _insert(self, root, key):
+        if root is None:
+            return TreeNode(key)
+        if key < root.key:
+            root.left = self._insert(root.left, key)
+        else:
+            root.right = self._insert(root.right, key)
+        return root
```

Comment:
`_insert` is a static method - it doesn't use any instance or class attributes. This applies also to other `_x` methods defined in `BinarySearchTree`

```suggestion
    def insert(self, key):
        self.root = BinarySearchTree._insert(self.root, key)

    @staticmethod
    def _insert(root, key):
        if root is None:
            return TreeNode(key)
        if key < root.key:
            root.left = BinarySearchTree._insert(root.left, key)
        else:
            root.right = BinarySearchTree._insert(root.right, key)
        return root
```